googletest_macro/src/lib.rs

```diff
@@ -327,3 +325,49 @@ pub fn derive(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
     }
     .into()
 }
+
+struct ToStringInput {
+    target: String,
+    length: usize
+}
+
+impl ToStringInput {
+    const DEFAULT_LENGTH_LIMIT: usize = 20;
```

Comment:
Is it essential to have a default length limit? I'd rather have users explicitly specify the length they want.

Reply:
I am actually thinking about getting rid of the customizable length, what do you think? The current issue is, for any expression, the macro would separate it token by token, and there is no way to solve it yet ([source](https://screenshot.googleplex.com/7WNwb9c6svwuBss)). For example:
`|x: i32| x + 1` becomes `| x : i32 | x + 1`. This might be okay, but what if we have `|x: Struct<Type>| pkg::pkg2::<Type>(x)`, which would become `| x : Struct < Type > | pkg :: pkg2 :: < Type > ( x )`, which makes this even harder to read.

If we can get rid of the customizable length limit and use a fixed default length, we can avoid this by converting the `TokenStream` to string directly, and then apply the shrinking method upon the stringified `TokenStream`. Or we could probably offer an environment variable to control the length if needed? What do you think? If you have some solution here, that will be great too!

Reply:
Or I have a hackier way to do this:

1. Convert the TokenStream to String directly without parsing (no format issue)
2. Find the last `,\s*\d+` pattern, and treat everything before it as an expression